gprofiler/metadata/metadata_collector.py

```diff
@@ -0,0 +1,35 @@
+import datetime
+import time
+from typing import Dict, Optional
+
+from gprofiler import __version__
+from gprofiler.metadata.cloud_metadata import get_static_cloud_instance_metadata
+from gprofiler.metadata.metadata_type import Metadata
+from gprofiler.metadata.system_metadata import get_static_system_info
+
+
+def get_static_metadata(spawn_time: Optional[float], run_args: Dict = None) -> Metadata:
+    if spawn_time is None:
```

Comment:
? why can it be `None`? Isn't it the spawn time of gProfiler?

Reply:
It could be used either way, if this method is called at spawn time

Reply:
There's exactly one call site:
```
        self._spawn_time = time.time()
        if collect_metadata and self._client is not None:
            self._static_metadata = get_static_metadata(spawn_time=self._spawn_time, run_args=run_args)
```